src/task.model.js

```diff
@@ -0,0 +1,83 @@
+import { v4 as uuidv4 } from 'uuid';
+import { getUsers, setUsers } from './userStorage.js';
+
+function mustBeInArray(id) {
+    return new Promise((resolve, reject) => {
+        const user = getUsers().find(u => u.id === id);
+
+        if (!user) {
+            reject({
+                message: 'User is not found',
+                status: 404
+            });
+        }
+        resolve(user);
+    });
+}
+
+function getNewId() {
+    return uuidv4();
+}
+
+export function getAutoSuggestUsers(loginSubstring, limit) {
+    const usersNumber = getUsers().length;
+    const usersLimit = usersNumber > limit ? Number(limit) : usersNumber;
+
+    return new Promise((resolve) => {
+        const suggestedUsers = getUsers()
+            .filter(user => user.login.includes(loginSubstring))
+            .sort((a, b) => {
+                if (a.login > b.login) return 1;
+                return a.login < b.login ? -1 : 0;
+            })
+            .slice(0, usersLimit);
+
+        resolve(suggestedUsers);
+    });
+}
+
+export function createUser(newUserData) {
+    return new Promise((resolve) => {
+        const id = { id: getNewId() };
+        const newUser = { ...id, ...newUserData, isDeleted: false };
```

Comment:
I see little sense in creation of `id` object and destructuring it after, probably just `const newUser = { ...newUserData, id: getNewId(), isDeleted: false };` would be better.